### ISSUE TITLE
vscode-tslint: error when using tslint 5.0.0-dev.0

### ISSUE THREAD

**Issue by akaRem** [Microsoft/vscode-tslint#180]
Hi!
I'm trying to setup env for working on TypeScript.
I've executed `git clone ..` and `npm install` and now I have a problem with vscode-tslint, which says `vscode-tslint: Cannot read tslint configuration - 'linter.findConfiguration is not a function'`

Here is what I have on my laptop (Mac w/ El Capitan)
```
# global
$ node -v
v7.7.4
$ tsc -v
Version 2.2.2
$ tslint -v
4.5.1

# local
$ ./node_modules/.bin/tsc -v
Version 2.3.0-dev.20170328
$ ./node_modules/.bin/tslint -v
5.0.0-dev.0

# vscode
$ code -v
1.10.2
8076a19fdcab7e1fc1707952d652f0bb6c6db331
$ code --list-extensions --show-versions | grep tslint
eg2.tslint@0.8.1

# git
$ git branch
* master
$ git rev-parse HEAD
0fd0903280d9c5090f18bcc22b7df7ca608857b1
$ git remote -v
origin	https://github.com/Microsoft/TypeScript.git (fetch)
origin	https://github.com/Microsoft/TypeScript.git (push)
# (I just cloned from original repo)
```

I tried to run without local modules at all (don't `npm install`), but then tslint can't find some custom rules. 
I tried to use local TS and local TSLint, but this didn't help too.

This is some kind of reopen of #123 #124

**Comment by egamma** [comment]
@akaRem I suspect this has to do with the locally installed tslint version `5.0.0-dev.0`. Can you pls try to reproduce this when you install tslint 4.5.0 locally?

**Comment by egamma** [comment]
What happens is that 5.0.0-dev.0 doesn't satisfy the semver check for > 4.0.0 and vscode-tslint thinks that this is an older tslint version where `findConfiguration` was available as a function.

**Comment by egamma** [comment]
published the fix as vscode-tslint 0.9.0

**Comment by akaRem** [comment]
@egamma I've upgraded and checked, now it doesn't fail on "Cannot read tslint configuration" but I still have some problems which are probably caused by TypeScript project itself.

jfyi https://github.com/Microsoft/TypeScript/issues/14917
